File: Practice/room_Java/app/src/main/java/com/example/room_java/TodoDao.java
package com.example.room_java;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface TodoDao {
    @Query("SELECT * FROM todo")
    LiveData<List<Todo>> getAll(); // getAll은 관찰 가능한 객체가 된다.

    @Insert
    void insert(Todo todo);

    @Update
    void update(Todo todo);

    @Delete
    void delete(Todo todo);

    @Query("DELETE FROM Todo")
    void deleteAll();

}

File: Practice/room_Kotlin/app/src/main/java/com/example/room_kotlin/TodoDao.kt
package com.example.room_kotlin

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface TodoDao {
    @Query("SELECT * FROM Todo")
    fun getAll(): LiveData<List<Todo>> //LiveData<> 로 감싸게 되면 실시간으로 관찰가능한 객체로 가능하게됨.

    @Insert
    fun insert(todo: Todo)

    @Update
    fun update(todo: Todo)

    @Delete
    fun delete(todo: Todo)
}